doc/source/release/1.14.0-notes.rst

```diff
@@ -49,49 +63,69 @@ New features
 
 `scipy.optimize` improvements
 =============================
+- `scipy.optimize.HessianUpdateStrategy` now also accepts square arrays for
+  ``init_scale``.
+- A new method, ``cobyqa``, has been added to `scipy.optimize.minimize` - this
+  is an interface for COBYQA (Constrained Optimization BY Quadratic
+  Approximations), a derivative-free optimization solver, designed to
+  supersede COBYLA, developed by the Department of Applied Mathematics, The
+  Hong Kong Polytechnic University.
+- There are some performance improvements in
+  `scipy.optimize.differential_evolution`.
+- `scipy.optimize.approx_fprime` now has subquadratic space complexity.
```

Comment:
or are we confident this is `O(n)` space now? I couldn't quite tell from skimming.

Reply:
No, I think it's performance needs improvement, it has a lot of internal overhead.
I gave the authors a PR on the cobyqa project that would greatly improve things but haven't heard back yet. This is one of the drawbacks of GitHub submodules, it's harder to fix things upstream. I requested that they merge the PR before the release, otherwise we may have to yank the PR that added the minimiser.

Reply:
Sorry, just realised that you were talking about approx_derivative/approx_fprime. Yes, the memory issue was fixed